cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetCache.java

```diff
@@ -276,18 +341,20 @@ public boolean equals(Object o) {
 
       DatasetCacheKey that = (DatasetCacheKey) o;
 
-      return Objects.equal(this.name, that.name) &&
+      return Objects.equal(this.namespace, that.namespace) &&
+        Objects.equal(this.name, that.name) &&
```

Comment:
we omit checking the accessType when performing lookups into the dataset cache, since we don't want to request another dataset instance, just because the accessType is different.
I believe @chtyim is changing this code, though, in his updates to read/write access for datasets.